File: C#/IsHappy.cs
public class Solution {
    public bool IsHappy(int n) {
        
        // Default
        if (n == 0)
        {
            return false;
        }
        
        int Sum = 0;
        List<int> Prev = new List<int>();
        
        while (n != 1)
        {
            int[] N   = n.ToString().Select(o=> Convert.ToInt32(o) - 48 ).ToArray();    
            n = N.Select(x => (int)Math.Pow(x, 2)).ToArray().Sum();   
            
            if (Prev.Contains(n))
            {
                return false;
            }
            
            Prev.Add(n);
            Console.WriteLine(n);
        }
        
        return true;
    }
}

File: Python/isHappy.py
class Solution:
    def isHappy(self, n: int) -> bool:

        test = n
        prev = []

        while True:

            digits = []

            while test > 0:
                digits.append(test % 10)

                test = test // 10

            test = 0

            for val in digits:
                test += pow(val, 2)

            if test in prev:
                return False

            if test == 1:
                return True

            prev.append(test)
